### ISSUE TITLE
Research approach to better text node updates rendering

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#13081]
Stems from #13051. There are at least two possible approaches to resolve that issue but with unclear feasibility and implications.

Let's check:

*   Is PoC quickly possible (2h max)
*   Test performance, effect on IME, typing, etc.
*   Whether tests pass, what else might go wrong.

**Comment by Reinmar** [comment]
@niegowski analyzed the new approach to rendering text nodes. It turns out to be a safe bet.

The next step is to remove `isAndroid` check in that specific place in the renderer to enable this sub-feature for all browsers. In addition to that we should add more tests.

**Comment by niegowski** [comment]
In this issue, I was researching how to reduce the number of view nodes marked to update by the renderer. The problem was that if we added some text to the existing text node, then the parent was marked as there are new children added to it. The result was a single text node with combined data but if the inserted text was wrapped with some attribute element then it was a real new child node for it. 

This "marking" of nodes is already a complicated part of the downcast writer. Also, marking an element to be updated in the DOM tree wasn't directly triggering the rerender of it, it was still verified in the renderer if some part of it needs an update. So I decided that it's more consistent with the way we handle changes of elements and while updating children of some view element, we are verifying if some text nodes should be updated and should not get replaced.
